server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -1068,6 +1069,38 @@ void compactionFailed(Map<ExternalCompactionId,KeyExtent> compactions) {
                     .findFirst().map(Map.Entry::getKey).orElse(null);
             LOG.debug("Unable to remove failed compaction {} {}", extent, ecid);
           }
+        } else {
+
+          // compactionFailed is called from the Compactor when either a compaction fails or
+          // is cancelled and it's called from the DeadCompactionDetector. This block is
+          // entered when the conditional mutator above successfully deletes an ecid from
+          // the tablet metadata. Remove compaction tmp files from the tablet directory
+          // that have a corresponding ecid in the name.
+          String dirName = ctx.getAmple().readTablet(extent, ColumnType.DIR).getDirName();
```

Comment:
ExternalCompaction_3_IT.testMergeCancelsExternalCompaction() was disabled.  I tried running it and it passed so I pushed a commit to enable it.  Did not bother with a PR since it was a single line change.

Reply:
Looked in tablet dir in each volume in 46e8c63

Reply:
Modified DoNothingExternalCompactor to create tmp file, added checks in the tests listed above, in 6e75f70